### ISSUE TITLE
Comment on files in repository

### ISSUE THREAD

**Issue by npcode** [dear-github/dear-github#161]
It would be nice if Github allows users to comment on not only commits but also files, as medium and google docs do.

Sometimes I want to leave a comment on README while reading it but it is a little bit difficult because I should find the proper commit in the history.

**Comment by stefnotch** [comment]
Or have issues linked to a specific file? Because, a comment would most likely be something like: "Fix that line of code" or "The readme is really hard to understand" or "+1 great code" (That should be avoided).

**Comment by npcode** [comment]
@stefanpenner Yes, that is the current workaround. Fortunately we can get a link to range of lines of a file in blame page.

**Comment by stefanpenner** [comment]
> @stefanpenner Yes, that is the current workaround. Fortunately we can get a link to range of lines of a file in blame page, even for a .md file.

wrong person i think.

**Comment by stefnotch** [comment]
Oh, yeah, we can link to lines in the code. 

Well, then, all that's missing is that the file displays the issues that are linked to it? (That would be just like comments...)

**Comment by npcode** [comment]
@stefanpenner Oops, sorry. I have modified my comment.

**Comment by npcode** [comment]
@stefnotch It sounds good. Additionally it would be nice if the link to the lines shows the preview like facebook and twitter. It helps to understand the context of the comment.

**Comment by stefnotch** [comment]
@npcode Great ideas! Include them in your post at the top?

**Comment by stefanpenner** [comment]
@npcode no worries, I do that all the time
